File: TouchNothing/Utilities/GuestNicknameGenerator.swift
import Foundation

enum GuestNicknameGenerator {
    private static let prefix = "guest_"
    private static let suffixLength = 6
    private static let charset = Array("abcdefghijklmnopqrstuvwxyz0123456789")

    static func generate() -> String {
        let suffix = (0..<suffixLength).map { _ in
            String(charset.randomElement() ?? "a")
        }.joined()
        return prefix + suffix
    }
}

File: app/src/main/java/com/touchnothing/app/util/GuestNicknameGenerator.kt
package com.touchnothing.app.util

object GuestNicknameGenerator {
    private const val PREFIX = "guest_"
    private const val SUFFIX_LENGTH = 6
    private const val CHARSET = "abcdefghijklmnopqrstuvwxyz0123456789"

    fun generate(): String {
        val suffix = (1..SUFFIX_LENGTH)
            .map { CHARSET.random() }
            .joinToString("")
        return PREFIX + suffix
    }
}
